website/payments/api/v2/admin/views.py

```diff
@@ -0,0 +1,130 @@
+import rest_framework.filters as framework_filters
+from django.apps import apps
+from django.http import Http404
+from django.utils.translation import gettext_lazy as _
+from oauth2_provider.contrib.rest_framework import IsAuthenticatedOrTokenHasScope
+from rest_framework import status, serializers
+from rest_framework.exceptions import PermissionDenied, ValidationError
+from rest_framework.generics import get_object_or_404
+from rest_framework.response import Response
+from rest_framework.reverse import reverse
+from rest_framework.settings import api_settings
+
+from payments import services, payables, NotRegistered
+from payments.api.v2 import filters
+from payments.api.v2.admin.serializers.payable_create import PayableCreateSerializer
+from payments.api.v2.admin.serializers.payable_detail import PayableSerializer
+from payments.api.v2.serializers import PaymentSerializer
+from payments.exceptions import PaymentError
+from payments.models import Payment, PaymentUser
+from thaliawebsite.api.v2.admin import (
+    AdminListAPIView,
+    AdminCreateAPIView,
+    AdminRetrieveAPIView,
+    AdminDestroyAPIView,
+    AdminUpdateAPIView,
+)
+
+
+class PaymentListCreateView(AdminListAPIView):
+    queryset = Payment.objects.all()
+    serializer_class = PaymentSerializer
+
+    required_scopes = ["payments:admin"]
+    filter_backends = (
+        framework_filters.OrderingFilter,
+        filters.CreatedAtFilter,
+        filters.PaymentTypeFilter,
+    )
+    ordering_fields = ("created_at",)
+
+
+class PaymentDetailView(AdminRetrieveAPIView, AdminDestroyAPIView):
+    queryset = Payment.objects.all()
+    serializer_class = PaymentSerializer
+    permission_classes = [IsAuthenticatedOrTokenHasScope]
+    required_scopes = ["payments:admin"]
```

Comment:
What exactly is this scope? It's not in the defined scopes in the settings (or is it something implicit with those AdminViews - I've never seen them)

Reply:
Have not added them to the settings yet. But the admin endpoints have a different scope than the regular endpoints.